### PR TITLE
6.5.0: Document JavaScript mutators

### LINKED ISSUE #3266: JavaScript mutators
### Description
JavaScript mutators will be added in 6.5.0. This is a new type of mutator.

This change also adds two new mutator attributes, `type` and `eval`.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-transform/mutators/
https://docs.sensu.io/sensu-go/latest/api/mutators/

### Corrections needed
Mutator reference should include JS mutator option and usage

### Additional context
https://github.com/sensu/sensu-go/issues/4354
https://github.com/sensu/sensu-go/pull/4361
[Eric's notes and examples](https://docs.google.com/document/d/1H2M9wkeUg3xYHSt9rfucsFHydIvfsv0QIjvq4DgizVA/edit#)
[Proposal section about JS mutators](https://docs.google.com/document/d/1rWDJoSFhpXRe9PUB8i7SVNA_eggJ6v5L8nJB7UaEkKc/edit#heading=h.mjnlema108za)

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds JavaScript mutator section and examples in mutator reference, along with `eval` and `type` attributes and description tables.

Also adds `type` attribute to examples in mutator API doc.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/3266

### CODE REVIEW

File: content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md [hunk 1]
```diff
@@ -94,14 +92,97 @@ spec:
...
+{
+  "type": "Mutator",
+  "api_version": "core/v2",
+  "metadata": {
+    "created_by": "admin",
+    "name": "js_mutator_1",
+    "namespace": "default"
+  },
+  "spec": {
+    "eval": "return JSON.stringify({"some info": "is here"});",
```
Comment (hillaryfraley) [inline content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md]: I think I need backslashes here for valid code

File: content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md [hunk 2]
```diff
@@ -119,6 +200,29 @@ As a result, executable scripts (like plugins) located in `/etc/sensu/plugins` w
 This allows `command` attributes to use “relative paths” for Sensu plugin commands (for example, `"command": "check-http.go -u https://sensuapp.org"`).
 {{% /notice %}}
 
+## JavaScript mutators
+
+Mutators that use JavaScript are an efficient alternative to pipe mutators, which fork a process on each invocation.
+JavaScript mutators are evaluated by the [Otto JavaScript VM][19] as JavaScript programs, which enables greater mutator throughput at scale.
+
+JavaScript mutators do not require you to return any value &mdash; you can either also mutate the events that are passed to the mutator.
```
Comment (echlebek) [inline content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md]: This sentence is a bit off

**Review by echlebek — APPROVED** [review]
Looks good, I just noticed some confusing wording in one of the sentences.

File: content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md (hunk 2 above)
Comment (echlebek) [reply content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md]: (line 208)

Comment (hillaryfraley) [reply content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md]: Yeah, that's messy. Thank you for catching it.

Comment (hillaryfraley) [inline content/sensu-go/6.5/observability-pipeline/observe-transform/mutators.md]:
```suggestion
JavaScript mutators do not require you to return any value &mdash; you can mutate the events that are passed to the mutator instead.
```

**Pull request merged**